src/routes/recipe.ts

```diff
@@ -38,4 +40,24 @@ router.get(
   validate,
   getRecipesLikeByUserId
 );
+router.delete(
+  "/:id",
+  protectedRoute,
+  getParamIdValidator(),
+  validate,
+  (req, res) => {
+    RecipeModel.deleteRecipeById({
+      recipeId: req.recipe.id,
+      userId: req.user.id,
+    })
+      .then((recipeData) => {
+        if (!recipeData) {
+          res.status(500).json({ message: "Did not find that recipe" });
+          return;
+        }
+        res.json(recipeData);
```

Comment:
This is good. I don't think this null check on `recipeData` is necessary because of the way we built the controller. If there were any issues finding the recipe or the user we throw an error, and the error should be caught in the `.catch` block below. I think you can just return the recipeData as a 200 response.